packages/medusa/src/api-v2/admin/regions/validators.ts

```diff
@@ -69,3 +69,21 @@ export class AdminGetRegionsParams extends extendedFindParamsMixin({
   @Type(() => AdminGetRegionsParams)
   $or?: AdminGetRegionsParams[]
 }
+
+export class AdminPostRegionsReq {
+  @IsString()
+  name: string
+
+  @IsString()
+  currency_code: string
+}
+
+export class AdminPostRegionsRegionReq {
```

Comment:
The naming here is a bit strange, looking from the code this is for updates only? Any reason we can't use simple names, like `UpdateRegionReq`?

Reply:
This goes far back, and I don't think we are actively using these names for much right now.

The idea was that payloads and query params should be derived from the request's path to ensure uniqueness and readability. This could prove useful in case we want to autogenerate stuff that uses these types. And it would allow the reader to always know exactly where a specific type is used, regardless of its location.

So for example: `POST /admin/products`

The types for the above endpoint would be:
- Payload: `AdminPostProductsReq`
- Query: `AdminPostProductsParams`
- General response: `AdminProductsRes`
- Delete response: `AdminProductsDeleteRes`

And you would know where this is used, even if placed in a separate folder or package.